servers/rendering/shader_compiler.cpp

```diff
@@ -1151,9 +1151,34 @@ String ShaderCompiler::_dump_node_code(const SL::Node *p_node, int p_level, Gene
 
 					if (!is_internal_func) {
 						for (int i = 0; i < shader->vfunctions.size(); i++) {
-							if (shader->vfunctions[i].name == vnode->name) {
-								func = shader->vfunctions[i].function;
-								break;
+							if (shader->vfunctions[i].rname == vnode->name) {
+								if (onode->arguments.size() == shader->vfunctions[i].function->arguments.size()) {
+									bool is_correct_args = true;
+
+									for (int j = 0; j < shader->vfunctions[i].function->arguments.size(); j++) {
+										const SL::DataType type_a = onode->arguments[j]->get_datatype();
+										const SL::DataType type_b = shader->vfunctions[i].function->arguments[j].type;
+
+										// Note: Checking for array size is unnecessary here.
+
+										if (type_a == type_b) {
+											if (type_a == SL::TYPE_STRUCT) {
+												if (onode->arguments[j]->get_datatype_name() != shader->vfunctions[i].function->arguments[j].struct_name) {
+													is_correct_args = false;
+													break;
+												}
+											}
+										} else {
+											is_correct_args = false;
+											break;
+										}
+									}
+
+									if (is_correct_args) {
+										func = shader->vfunctions[i].function;
+										break;
+									}
+								}
 							}
```

Comment:
```suggestion
							if (shader->vfunctions[i].rname == vnode->name && onode->arguments.size() == shader->vfunctions[i].function->arguments.size()) {
                                bool is_correct_args = true;

								for (int j = 0; j < shader->vfunctions[i].function->arguments.size(); j++) {
									const SL::DataType type_a = onode->arguments[j]->get_datatype();
									const SL::DataType type_b = shader->vfunctions[i].function->arguments[j].type;

									// Note: Checking for array size is unnecessary here.

									if (type_a == type_b) {
										if (type_a == SL::TYPE_STRUCT && onode->arguments[j]->get_datatype_name() != shader->vfunctions[i].function->arguments[j].struct_name) {
											is_correct_args = false;
                                            break;
										}
									} else {
										is_correct_args = false;
										break;
									}
								}

								if (is_correct_args) {
									func = shader->vfunctions[i].function;
									break;
								}
                            }
```
Merge conditions to avoid unnecessary indentation

Reply:
Thanks, done